include/sparrow_v01/layout/variable_size_binary_array.hpp

```diff
@@ -0,0 +1,594 @@
+                                                    impl::get_inner_reference_t<array_inner_types<L>, is_const>>
+    {
+    public:
+
+        using self_type = variable_size_binary_value_iterator<L, is_const>;
+        using base_type = iterator_base<
+            self_type,
+            mpl::constify_t<typename array_inner_types<L>::inner_value_type, is_const>,
+            std::contiguous_iterator_tag,
+            impl::get_inner_reference_t<array_inner_types<L>, is_const>>;
+        using reference = typename base_type::reference;
+        using difference_type = typename base_type::difference_type;
+        using size_type = typename array_inner_types<L>::size_type;
+        using layout_type = mpl::constify_t<L, is_const>;
+
+        variable_size_binary_value_iterator() noexcept = default;
+        variable_size_binary_value_iterator(layout_type* layout, size_type index);
+
+    private:
+
+        reference dereference() const;
+        void increment();
+        void decrement();
+        void advance(difference_type n);
+        difference_type distance_to(const self_type& rhs) const;
+        bool equal(const self_type& rhs) const;
+        bool less_than(const self_type& rhs) const;
+
+        layout_type* p_layout = nullptr;
+        difference_type m_index;
+
+        friend class iterator_access;
+    };
+
+    /**
+     * Implementation of reference to inner type used for layout L
+     *
+     * @tparam L the layout type
+     */
+    template <class L>
+    class variable_size_binary_reference
+    {
+    public:
+
+        using self_type = variable_size_binary_reference<L>;
+        using value_type = typename L::inner_value_type;
+        using reference = typename L::inner_reference;
+        using const_reference = typename L::inner_const_reference;
+        using size_type = typename L::size_type;
+        using difference_type = std::ptrdiff_t;
+        using iterator = typename L::value_iterator;
+        using const_iterator = typename L::const_value_iterator;
+        using offset_type = typename L::offset_type;
+
+        variable_size_binary_reference(L* layout, size_type index);
+        variable_size_binary_reference(const variable_size_binary_reference&) = default;
+        variable_size_binary_reference(variable_size_binary_reference&&) = default;
+
+        template <std::ranges::sized_range T>
+            requires mpl::convertible_ranges<T, typename L::inner_value_type>
+        self_type& operator=(T&& rhs);
+
+        // This is to avoid const char* from begin caught by the previous
+        // operator= overload. It would convert const char* to const char[N],
+        // including the null-terminating char.
+        template <class U = typename L::inner_value_type>
+            requires std::assignable_from<U&, const char*>
+        self_type& operator=(const char* rhs);
+
+        size_type size() const;
+
+        // iterator begin();
```

Comment:
I commented the "mutable" parts. Tell me if you prefer that I just remove the code

Reply:
Let's keep it if it don't bother you.